### PR TITLE
Feature needs and quests

### PR DESCRIPTION
**Pull request by YanaSuprun**
-Changed design
-Added functionality
-Server requests implemented partually (because of problems with back. Will be finalized today)
-Added some tests
-Some changes according to v2 api
Снимок экрана от 2019-11-18 12-14-45
Снимок экрана от 2019-11-18 12-15-19
Снимок экрана от 2019-11-18 02-26-32

### CODE REVIEW

File: src/app/donation/needs/needs-item-details/needs-item-details.component.ts [hunk 1]
```diff
@@ -11,14 +14,83 @@ import { Observable } from 'rxjs';
...
+    this.needService.putNeedData(this.needId, newData)
+    .subscribe(_ => {
+      this.notifierService.showNotice('Changes have been saved!', 'success');
+    });
+  }
+
+  public onDelete(): void {
+    this.needService.deleteNeed(this.needId)
+      .subscribe(_ => {
+        this.notifierService.showNotice(`Need ${this.need.itemName} was deleted!`, 'error');
```
Comment (ksenya-ta) [inline src/app/donation/needs/needs-item-details/needs-item-details.component.ts]: use error only for errors, if you successfully deleted something it also should use success notification (green one)

File: src/app/donation/quests/quests-item/quests-item.component.ts [hunk 2]
```diff
@@ -8,4 +8,8 @@ import { Quest } from '../models/quest.interface';
 })
 export class QuestsItemComponent {
   @Input() quest: Quest;
+
+  public onExecute() {
+    // changing status by put method will be here
+  }
```
Comment (ksenya-ta) [inline src/app/donation/quests/quests-item/quests-item.component.ts]: please once you get to implement it do not perform any put requests here, just notify the parent using appropriate outputs so he can call what you need in already injected service, keep this component as dumb one

Comment (YanaSuprun) [reply src/app/donation/quests/quests-item/quests-item.component.ts]: yes, of course. It's just a confuse in my explanation.
Thx)

File: src/app/donation/needs/needs-item-details/needs-item-details.component.ts (hunk 1 above)
Comment (YanaSuprun) [reply src/app/donation/needs/needs-item-details/needs-item-details.component.ts]: ok. thx

**Pull request merged**